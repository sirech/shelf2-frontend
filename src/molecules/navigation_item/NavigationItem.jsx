// @flow

import React from 'react'
import { Badge, ListGroupItem } from 'reactstrap'
import { LinkContainer } from 'react-router-bootstrap'

const NavigationItem = ({year, count}: { year: number, count: number}) => (
  <LinkContainer to={`/books/${year}`}>
    <ListGroupItem tag='a'>
      <span>{year}</span >
      <Badge pill color='warning' className='ml-auto'>{count}</Badge>
    </ListGroupItem>
  </LinkContainer>
)

export default NavigationItem
