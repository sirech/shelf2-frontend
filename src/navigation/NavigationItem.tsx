import React from 'react'
import { Badge, ListGroupItem } from 'reactstrap'
import { NavigateFunction, useLocation, useNavigate } from 'react-router'

const safeNavigate = (
  event: React.MouseEvent<HTMLElement>,
  navigate: NavigateFunction,
  to: string
) => {
  const isModifiedEvent = () =>
    !!(event.metaKey || event.altKey || event.ctrlKey || event.shiftKey)

  if (
    !event.defaultPrevented && // onClick prevented default
    event.button === 0 && // ignore right clicks
    !isModifiedEvent() // ignore clicks with modifier keys
  ) {
    event.preventDefault()

    navigate(to)
  }
}

const NavigationItem = ({ year, count }: { year: number; count: number }) => {
  const navigate = useNavigate()
  const { pathname } = useLocation()
  const link = `/books/${year}`

  return (
    <ListGroupItem
      action
      tag="a"
      className="d-flex justify-content-between align-items-center"
      active={link === pathname}
      onClick={(event) => safeNavigate(event, navigate, link)}
    >
      <span>{year}</span>
      <Badge pill color="warning" className="ml-auto text-dark">
        {count}
      </Badge>
    </ListGroupItem>
  )
}

export default NavigationItem
