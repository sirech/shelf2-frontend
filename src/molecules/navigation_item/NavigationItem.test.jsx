import React from 'react'
import { fullRender } from '../../test'
import NavigationItem from './NavigationItem'

describe('components', () => {
  describe('NavigationItem', () => {
    it('renders correctly', () => {
      const { component } = fullRender(<NavigationItem year='2016' count='2' />)
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
