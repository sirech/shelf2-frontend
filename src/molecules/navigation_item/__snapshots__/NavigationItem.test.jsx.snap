// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components NavigationItem renders correctly 1`] = `
<li
  activeClass="active"
  className="list-group-item"
  href="/books/2016"
  onClick={[Function]}
  style={undefined}
>
  <span>
    2016
  </span>
  <span
    className="ml-auto badge badge-warning badge-pill"
  >
    2
  </span>
</li>
`;
